### DISCUSSION TITLE
Jenkins download path

### DISCUSSION THREAD

**Discussion by mwonderful-idbydna** [ElSnoMan/pyleniumio#227] [Ideas]
Hi! Apologies for not doing diligence if this has been already suggested. 

Jenkins needs a download path relative to the build directory but Chrome ignores relative paths. This gets set in pylenium.json:

`    
"experimental_options": [
        {"prefs": {"download.default_directory": "downloads" }}
 ],
`

Chrome ignores relative paths. Therefore we added a function in conftest.py to prepend the current working directory to the relative path so each Jenkins plan uses a directory relative to that build directory. Add a call this function in py_config inside conftest.py to set a build specific downloadd path. 

`

    def env_config(config):
    """Environmental configurations"""
    
    expopts = config.driver.experimental_options
        
    if ((expopts is None) or (len(expopts) == 0)):
        raise AssertionError("pylenium.json missing download.default_directory")

    download_directory = None
    new_expopts = []        
    for item in expopts:
        if (isinstance(item, dict) and ('prefs' in item)):
            if not 'download.default_directory' in item['prefs']:
                raise AssertionError("pylenium.json missing download.default_directory")
            ddir = item['prefs']['download.default_directory']
            ddir = os.path.join(os.getcwd(), '..', ddir)
            ddir = os.path.abspath(ddir)
            if not os.path.isdir(ddir):
                raise IOError("%s download directory now found." %
                              (ddir, ))

            item['prefs']['download.default_directory'] = ddir
            download_directory = ddir
            new_expopts.append(item)
        else:
            new_expopts.append(item)

    config.driver.experimental_options = new_expopts
                    
    if download_directory is None:
        raise AssertionError("pylenium.json missing download.directory_directory")
    print("%s => new download directory." % (download_directory, ))
`

**Comment by ElSnoMan** [comment]
Is this only for Chrome and Jenkins or does this work for other browsers too?

**Reply by mwonderful-idbydna** [reply]
Sorry, I have only tested Chrome and Jenkins.
